src/transactions/CategoryFilter.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { useSelector } from 'react-redux';
+
+/**
+ * Displays a select dropdown of categories as options to filter by and
+ * a clear button to reset the dropdown.
+ *
+ * @component
+ */
+function CategoryFilter({ categoryId, handleFilter }) {
+  const onSelectChanged = ({ target: { value } }) => {
+    handleFilter({ categoryId: value });
+  };
+
+  const onClear = () => {
+    handleFilter({ categoryId: null });
+  };
+
+  let selectedValue = categoryId;
+  if (!categoryId) {
+    selectedValue = '';
+  }
+
+  const categoryList = useSelector((state) => state.categories.entities);
+
+  return (
+    <div className="row">
+      <div className="col col-3 mr-2">
+        <select className="form-select" value={selectedValue} name="categoryFilter" onChange={onSelectChanged}>
+          <option value="">Filter by Category</option>
+          {categoryList.map((category) => (
+            <option key={category.id} value={category.id} name={category.name}>
```

Comment:
Teeny thing I just noticed, you can remove the name attribute here. options don't have name attributes.